### PR TITLE
dragons3d iOS/Android merge

### PR DESCRIPTION
**Pull request by ojcastillo**

### CODE REVIEW

File: PubNub/Data/PNAES.m [hunk 1]
```diff
@@ -6,6 +6,9 @@
 #import "PNAES.h"
 #import <CommonCrypto/CommonCryptor.h>
 #import <CommonCrypto/CommonHMAC.h>
+#ifdef PGDROID
+//#import <CommonCrypto/CommonCryptoError.h>
```
Comment (mjk3979) [inline PubNub/Data/PNAES.m:10]: Why is this here if it's commented out?

File: PubNub/Data/PNAES.m [hunk 2]
```diff
@@ -356,8 +359,12 @@ + (NSError *)errorFor:(CCCryptorStatus)status {
             errorCode = kPNAESInsufficientMemoryError;
             break;
         case kCCDecodeError:
+#ifdef PGDROID
+#   warning HC SVNT DRACONES! : CommonCrypto is not fully implemented on PGDroid ...
```
Comment (mjk3979) [inline PubNub/Data/PNAES.m:363]: What does this do?

File: PubNub/Network/PNNetwork.m [hunk 3]
```diff
@@ -18,6 +18,9 @@
 #import "PNConstants.h"
 #import "PNHelpers.h"
 
+#ifdef PGDROID
+    #import <UIKit/UIKit.h>
```
Comment (mjk3979) [inline PubNub/Network/PNNetwork.m:22]: Why do we need UIKit here?

File: PubNub/Core/PubNub+Core.m [hunk 4]
```diff
@@ -16,6 +16,9 @@
 #import "PNNetwork.h"
 #import "PNHelpers.h"
 
+#ifdef PGDROID
+#import <UIKit/UIKit.h>
```
Comment (mjk3979) [inline PubNub/Core/PubNub+Core.m:20]: Why do we need UIKit?

File: PubNub/Data/PNAES.m (hunk 1 above)
Comment (mauiaaron) [inline PubNub/Data/PNAES.m:10]: That header is missing in Apportable platform.  But probably shouldn't be...

File: PubNub/Data/PNAES.m (hunk 2 above)
Comment (mauiaaron) [inline PubNub/Data/PNAES.m:363]: Just a simple compiler warning that our security/crypto layer is severely lacking on Android.

File: PubNub/Core/PubNub+Core.m (hunk 4 above)
Comment (mauiaaron) [inline PubNub/Core/PubNub+Core.m:20]: Fix a few more compiler errors : 

```
/Users/aaron/A/dragons3d/Externals/pgengine/Externals/PubNub/PubNub/Core/PubNub+Core.m:219:41: error: use of undeclared identifier 'UIApplicationWillEnterForegroundNotification'
                                   name:UIApplicationWillEnterForegroundNotification object:nil];
                                        ^
/Users/aaron/A/dragons3d/Externals/pgengine/Externals/PubNub/PubNub/Core/PubNub+Core.m:221:41: error: use of undeclared identifier 'UIApplicationDidEnterBackgroundNotification'
                                   name:UIApplicationDidEnterBackgroundNotification object:nil];
                                        ^
```

It's likely that something should be including UIKit/UIKit.h by default on Apportable but isn't for some reason.

File: PubNub/Data/PNAES.m (hunk 2 above)
Comment (mauiaaron) [inline PubNub/Data/PNAES.m:363]: In particular `kCCOverflow` and `kCCRNGFailure` do not exist in platform.  The ideal fix will be to bring in the new Apple headers (into Apportable) and iterate on supporting them.  This is likely to be a large platform task.

File: PubNub/Network/PNNetwork.m (hunk 3 above)
Comment (mauiaaron) [inline PubNub/Network/PNNetwork.m:22]: Fix a bunch of compiler errors:

```
/Users/aaron/A/dragons3d/Externals/pgengine/Externals/PubNub/PubNub/Network/PNNetwork.m:870:16: error: use of undeclared identifier 'UIDevice'
    device = [[UIDevice currentDevice] model];
               ^
/Users/aaron/A/dragons3d/Externals/pgengine/Externals/PubNub/PubNub/Network/PNNetwork.m:871:29: error: use of undeclared identifier 'UIDevice'
    NSString *osVersion = [[UIDevice currentDevice] systemVersion];
```

File: PubNub/Data/PNAES.m (hunk 1 above)
Comment (mauiaaron) [inline PubNub/Data/PNAES.m:10]: Actually we should just remove this.  Done in 7ef9eb4b52c7588239d5381e667b748e43fe78b5

**Comment by mjk3979** [comment]
Changes lgtm
